File: packages/rust/crates/oxbitnet-swift/Sources/OxBitNet/GenerateOptions.swift
import Foundation

/// Options for text generation.
public struct GenerateOptions: Sendable {
    public var maxTokens: Int
    public var temperature: Float
    public var topK: Int
    public var repeatPenalty: Float
    public var repeatLastN: Int

    public init(
        maxTokens: Int = 256,
        temperature: Float = 1.0,
        topK: Int = 50,
        repeatPenalty: Float = 1.1,
        repeatLastN: Int = 64
    ) {
        self.maxTokens = maxTokens
        self.temperature = temperature
        self.topK = topK
        self.repeatPenalty = repeatPenalty
        self.repeatLastN = repeatLastN
    }
}

File: packages/rust/crates/oxbitnet-csharp/src/OxBitNet/GenerateOptions.cs
namespace OxBitNet
{
    /// <summary>
    /// Options for text generation.
    /// </summary>
    public sealed class GenerateOptions
    {
        /// <summary>Maximum number of tokens to generate.</summary>
        public int MaxTokens { get; set; } = 256;

        /// <summary>Sampling temperature.</summary>
        public float Temperature { get; set; } = 1.0f;

        /// <summary>Top-k sampling parameter.</summary>
        public int TopK { get; set; } = 50;

        /// <summary>Repetition penalty.</summary>
        public float RepeatPenalty { get; set; } = 1.1f;

        /// <summary>Window size for repetition penalty.</summary>
        public int RepeatLastN { get; set; } = 64;
    }
}
